### ISSUE TITLE
Helm chart doesn't support the experimental object

### ISSUE THREAD

**Issue by briggsy87** [DataDog/helm-charts#529]
https://docs.datadoghq.com/tracing/setup_overview/open_standards/#otlp-ingest-in-datadog-agent Describes the steps to configure the agent to ingest OLTP.

**Describe what happened:**
Helm chart successfully deploys, but does not pass through the variables passed.

**Describe what you expected:**
To allow configuration of the experimental config block as part of the helm chart.

**Steps to reproduce the issue:**
`helm repo add datadog https://helm.datadoghq.com  `

`helm install -n logging ddagent --set datadog.apiKey=$datadog_apikey datadog/datadog --set targetSystem=linux --set datadog.dogstatsd.useHostPort=true --set experimental.otlp.receiver.protocols.http.endpoint=0.0.0.0:4317 --create-namespace`

**Additional environment details (Operating System, Cloud provider, etc):**
Kubernetes v1.21.7

**Comment by briggsy87** [comment]
Hi @clamoriniere,

Sorry, just so I am clear, I can't use the experimental block but I CAN set both of these other variables through helm?

I had tried a number of combinations to do so, but none seem to work:
```
helm upgrade -n logging ddagent datadog/datadog \
  --set datadog.apiKey=$datadog_apikey  \
  --set targetSystem=linux \
  --set datadog.dogstatsd.useHostPort=true \
  --set "agents.containers.agent.ports=[4318]" \
  --set "agents.containers.traceAgent.env[0].name=DD_OTLP_HTTP_PORT" \
  --set "agents.containers.traceAgent.env[0].value=4318" \
  --create-namespace
```

I am sure it is just me, but maybe you could provide an example helm command that sets this up? I mean, I understand not having helm configure the experimental block, but it would still be great if the documentation covered the workaround for those who are using your official helm chart as their preferred installation method.

**Comment by clamoriniere** [comment]
Hi @briggsy87 

Sorry, I gave you some wrong information on my previous message. it is the `agent` container that needs to expose the port 4318.

the following values.yaml file should provide the correct configuration

```yaml
datadog:
  dogstatsd:
    useHostPort: true
  apm:
    portEnabled: true

agents:
  containers:
    agent:
      env:
      - name: DD_OTLP_HTTP_PORT
        value: "4318"
      ports:
      - containerPort: 4317
        hostPort: 4317
        name: otplport
        protocol: TCP
```

**Comment by mkorejo** [comment]
Hi @clamoriniere, beyond the steps above, do we need to make any adjustments to the cluster-agent Service to be able to configure `OTEL_EXPORTER_OTLP_ENDPOINT` to receive data on `DD_OTLP_HTTP_PORT`? I'm referencing [this](https://docs.datadoghq.com/tracing/setup_overview/open_standards/#example-kubernetes-opentelemetry-application-configuration).

**Comment by clamoriniere** [comment]
No the cluster-agent service does not receive otel traffic. Only the agent deployed on each node do.
In that case any service is use, but it is received directly by an hostport.

**Comment by mkorejo** [comment]
Thanks for the support thus far, unfortunately though still having issues. Our goal is to use DD agent to collect both logs and OTLP trace and metrics (no separate OpenTelemetry collector).

My current Helm values are:
```
  values:
    datadog:
      apiKeyExistingSecret: datadog-api-key
      clusterName: techops-k8s-sandbox
      dogstatsd:
        useHostPort: true
      apm:
        portEnabled: true
      logs:
        enabled: true
        containerCollectAll: true
        autoMultiLineDetection: true
      containerExcludeLogs: "image:bitnami/mongodb* kube_namespace:datadog kube_namespace:kube-* kube_namespace:vault"
      processAgent:
        processCollection: false
      site: us3.datadoghq.com
    agents:
      image:
        tag: 7.35.0-rc.6
      containers:
        agent:
          env:
          - name: DD_OTLP_GRPC_PORT
            value: "4317"
          ports:
          - containerPort: 4317
            hostPort: 4317
            name: otlpport
            protocol: TCP
        traceAgent:
          env:
          - name: DD_OTLP_GRPC_PORT
            value: "4317"
    targetSystem: linux
```

With this configuration, I see these messages in `agent` log:
```
❯ klog infra-datadog-2pjft -c agent | grep otlp
2022-03-21 18:54:03 UTC | CORE | WARN | (pkg/util/log/log.go:587 in func1) | OTLP ingest configuration is now stable and has been moved out of the "experimental" section. This section will be removed in the 7.37 Datadog Agent release. Please use the "otlp_config" section instead.The DD_OTLP_GRPC_PORT and DD_OTLP_HTTP_PORT environment variables will also be removed in 7.37; set the full endpoint instead.
2022-03-21 18:54:03 UTC | CORE | WARN | (pkg/util/log/log.go:592 in func1) | failed to get configuration value for key "experimental.otlp": unable to cast <nil> of type <nil> to map[string]interface{}
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/exporters_builder.go:255 in buildExporter) | kind:exporter,name:otlp | Exporter was built.
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/receivers_builder.go:226 in attachReceiverToPipelines) | kind:receiver,name:otlp,datatype:traces | Receiver was built.
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/receivers_builder.go:226 in attachReceiverToPipelines) | kind:receiver,name:otlp,datatype:metrics | Receiver was built.
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/exporters_builder.go:40 in Start) | kind:exporter,name:otlp | Exporter is starting...
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/exporters_builder.go:48 in Start) | kind:exporter,name:otlp | Exporter started.
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/receivers_builder.go:68 in StartAll) | kind:receiver,name:otlp | Receiver is starting...
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/receiver/otlpreceiver/otlp.go:69 in startGRPCServer) | kind:receiver,name:otlp | Starting GRPC server on endpoint 0.0.0.0:4317
2022-03-21 18:54:05 UTC | CORE | INFO | (collector@v0.44.0/service/internal/builder/receivers_builder.go:73 in StartAll) | kind:receiver,name:otlp | Receiver started.
```

And in my deployment:
```
      env:
        - name: HOST_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.hostIP
        - name: OTEL_EXPORTER_OTLP_ENDPOINT
          value: http://$(HOST_IP):4317
```


Anything I'm missing that you can think of?

**Comment by mkorejo** [comment]
Nothing missing, we figured it out!

**Comment by clamoriniere** [comment]
Hi @mkorejo 
Happy to know that it is working for you. 

Could you elaborate what was the issue?

**Comment by mkorejo** [comment]
In our case, we needed to add the environment variable `OTLP_COLLECTOR` to our Deployment manifests with value `http://$(HOST_IP):4317`.

So, we swapped `OTEL_EXPORTER_OTLP_ENDPOINT` for `OTLP_COLLECTOR` based on how the OpenTelemetry SDK was being initialized in our code.

**Comment by tomerfriedman** [comment]
@guizmaii what's the url you used in the OTLPTraceExporter? (in yout application code)

**Comment by guizmaii** [comment]
@tomerfriedman in your app helm chart you need to add something like this:

```yaml
      env:
        - name: HOST_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.hostIP
        - name: OTLP_ENDPOINT
          value: http://$(HOST_IP):4317 # or 4318 if you use HTTP
```

and then use this `OTLP_ENDPOINT` environment variable as your URL

**Comment by guizmaii** [comment]
@mkorejo How did you solve your issue? I have configured my Datadog Agent as explained here https://github.com/DataDog/helm-charts/issues/529#issuecomment-1098166788, then configure the Helm Chart of my app as explained here https://github.com/DataDog/helm-charts/issues/529#issuecomment-1098817576.

I can see in my app that the value of `OTLP_ENDPOINT` is correctly passed to my `OtlpGrpcSpanExporter` instance but I don't see any traces in DD 🤔

**Comment by guizmaii** [comment]
> I don't see any traces in DD

Problem solved. Problem and solution explained here: https://github.com/DataDog/datadog-agent/issues/11664#issuecomment-1100231338

**Comment by adriangb** [comment]
@clamoriniere I'm trying to get this working and running into issues. Here's my (relevant) values:

```yaml
datadog:
  logsEnabled: true
  logsConfigContainerCollectAll: true
  apm:
    portEnabled: true
agents:
  containers:
    agent:
      env:
        - name: DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT
          value: "0.0.0.0:4318"
      ports:
        - containerPort: 4318
          hostPort: 4318
          name: otlphttpport
          protocol: TCP
    traceAgent:
      env:
        - name: DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT
          value: "0.0.0.0:4318"
```

I'm using `v2.32.2` of the chart and `v7.35.0` of the agent.

In my agent logs I see:

```
INFO | (collector@v0.44.0/receiver/otlpreceiver/otlp.go:87 in startHTTPServer) | kind:receiver,name:otlp | Starting HTTP server on endpoint 0.0.0.0:4318
```

And some other stuff about the server starting up.
I don't see anything in the traceAgent logs.

When I send a span I get back a 200 OK. If I remove the port from my values I get a connection error, so I know that the spans are being sent and making it to the agent.

It's not the issue referenced in https://github.com/DataDog/helm-charts/issues/529#issuecomment-1100232079, I am configuring the env:

```python
import os
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import (
   SimpleSpanProcessor,
)
from opentelemetry.exporter.otlp.proto.http.trace_exporter import (  # type: ignore
    OTLPSpanExporter,
)
from opentelemetry.sdk.resources import (  # type: ignore
    DEPLOYMENT_ENVIRONMENT,
    SERVICE_NAME,
    SERVICE_VERSION,
    Resource,
)
from opentelemetry.exporter.datadog import DatadogExportSpanProcessor, DatadogSpanExporter

provider = TracerProvider(
   resource=Resource(
      attributes={
         SERVICE_NAME: "test",
         SERVICE_VERSION: "0.0.0",
         DEPLOYMENT_ENVIRONMENT: "dev",
      }
   )
)
host_ip = os.environ["HOST_IP"]
exp1 = SimpleSpanProcessor(OTLPSpanExporter(f"http://{host_ip}:4318/v1/traces"))
exp2 = SimpleSpanProcessor(
    DatadogSpanExporter(
        agent_url=f"http://{host_ip}:8126", service="my-helloworld-service"
    )
)
provider.add_span_processor(exp1)
# provider.add_span_processor(exp2)
trace.set_tracer_provider(provider)
tracer = trace.get_tracer(__name__)
with tracer.start_as_current_span("span123"):
        print("Hello world!")
```

Also if I uncomment `provider.add_span_processor(exp2)` (i.e. enable exporting via Datadog's exporter, which is now deprecated it seems) I _do_ get traces showing up in DataDog and I see log entries in the `traceAgent` logs, so I'm fairly certain this isn't an issue of misconfiguring the resource.

My first question is: how is communicating happening here? Is it `app -> agent -> traceAgent -> datadog.com`? If so, why does the `traceAgent` require the `DD_OTLP_HTTP_PORT` / `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT ` env var if it's not receiving OTLP traces directly (presumably it's receiving them via the internal port I'm seeing in the logs)?

**Comment by duxing** [comment]
@adriangb I'm running into the same problem with a very similar set up compared to yours.

spent a good amount of time digging into datadog-agent source code and I believe you're right on the traffic flow.
 - `app` sends traces in `otlp` over `grpc` to `agent` on `4317`
 - `agent` export received traces over `grpc` to `trace-agent` on internal port `5003` [ref](https://github.com/DataDog/datadog-agent/blob/7.35.0/pkg/trace/api/otlp.go#L82)
 - `trace-agent` export traces to datadog backend.

I've verified with `tcpdump` on port `4317` or `5003`: `tcpdump -nnA -i any "port 4317 or port 5003" -s 0`.
when trace is sent from my app, I saw the right trace in `otlp` received on port `4317` but nothing was received on port `5003`.

set logs to trace and didn't see error / failures in either `agent.log` or `trace-agent.log`
